File: src/main/Java/boj_step4/boj10811.java
import java.io.*;
import java.util.StringTokenizer;
import java.util.stream.IntStream;

public class boj10811 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
        int n = Integer.parseInt(tokenizer.nextToken());
        int m = Integer.parseInt(tokenizer.nextToken());
        int[] basket = IntStream.range(1, n + 1).toArray();
        while (m-- > 0) {
            tokenizer = new StringTokenizer(reader.readLine());
            int i = Integer.parseInt(tokenizer.nextToken()) - 1;
            int j = Integer.parseInt(tokenizer.nextToken()) - 1;
            int partition = (j - i) / 2;
            while (partition-- >= 0) {
                int tmp = basket[i];
                basket[i++] = basket[j];
                basket[j--] = tmp;
            }
        }
        for (int ball: basket) writer.append(ball + " ");
        writer.close();
    }
}

File: src/main/Kotlin/others/boj10811.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    var (n, m) = r.readLine()!!.split(" ").map { it.toInt() }
    var basket = mutableListOf(0)
    for (i in 1 .. n) basket.add(i)
    repeat(m) {
        var (i, j) = r.readLine()!!.split(" ").map { it.toInt() }
        repeat((j - i)/2 + 1) {
            val tmp = basket[i]
            basket[i++] = basket[j]
            basket[j--] = tmp
        }
    }
    for (i in 1 until basket.size) w.write("${basket[i]} ")
    w.close()
}